src/modules/systemlib/battery.cpp

```diff
@@ -239,12 +230,8 @@ Battery::computeScale()
 	// reusing capacity calculation to get single cell voltage before drop
 	const float bat_v = _param_v_empty.get() + (voltage_range * _remaining_voltage);
 
-	_scale = _param_v_full.get() / bat_v;
-
-	if (_scale > 1.3f) { // Allow at most 30% compensation
-		_scale = 1.3f;
+	// Allow at most 30% compensation
+	// Shouldn't ever be more than the power at full battery
+	_scale = math::constrain(_param_v_full.get() / bat_v, 1.0f, 1.3f);
```

Comment:
I think not. But I'm not familiar enough with this code to say how big a problem it really is. Generally I think these values should be checked at the source, not the destination where they're used, because the wrong values then already propagated partly through the system and it's harder to make a good decision which value to use.

Adding checks to constrain might be a solution, to provide some stronger guarantees (I know how hard it can be to track down NaN's). The question is which value to use on NaN, the lower or upper limit?